### PR TITLE
Add Nickel syntax highlighting

### PR DESCRIPTION
**Pull request by szlend**
- Adds [tree-sitter-nickel](https://github.com/nickel-lang/tree-sitter-nickel) ([nickel-lang.org](https://nickel-lang.org/))
- Adds highlights and injections queries

Release Notes:

- Added Nickel syntax highlighting.

### CODE REVIEW

**Comment by szlend** [comment]
@cla-bot check

**Comment by SomeoneToIgnore** [comment]
Please also add a new docs entry to the https://github.com/zed-industries/zed/tree/main/docs/src/languages list

File: crates/zed/src/languages/nickel/injections.scm [hunk 1]
```diff
@@ -0,0 +1,3 @@
+(annot_atom doc: (static_string)
+            @injection.content
+            (#set! injection.language "markdown"))
```
Comment (szlend) [inline crates/zed/src/languages/nickel/injections.scm:3]: I don't think injections are working. Let me play around with this.

Comment (szlend) [reply crates/zed/src/languages/nickel/injections.scm]: I think I fixed it. Adding a new line inside a doc coment after a `#` no longer inserts another `#` in the new line (it's markdown, not a code comment).

File: crates/zed/src/languages/nickel/highlights.scm [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  "force"
+  "not_exported"
+] @keyword
+
+"fun" @keyword.function
+
+"import" @include
+
+[ "if" "then" "else" ] @conditional
+"match" @conditional
```
Comment (szlend) [inline crates/zed/src/languages/nickel/highlights.scm:25]: Not highlighted.

Comment (szlend) [reply crates/zed/src/languages/nickel/highlights.scm]: Replaced `@conditional` -> `@keyword`
Replaced `@include` -> `@keyword`

**Comment by maxbrunsfeld** [comment]
@szlend Thanks for the PR. I think that for right now, we need to limit how many languages we bundle into Zed because of compile time issues. Because Nickel is a bit more obscure of a language, I'd like to hold off on adding it, for the sake of our app's build time.

We are actively working on the ability for Zed to load arbitrary Tree-sitter parsers via Wasm files. It is one of my top priorities at the moment. Once that happens, we'll be able to have a Nickel plugin, and we plan to automatically surface available plugins when users are editing unsupported file extensions. You'll be able to package the exact code that you wrote here into an extension.

Here is a tracking issue for our support of language extensions: https://github.com/zed-industries/zed/issues/7096.

**Comment by szlend** [comment]
No worries, makes sense.

**Pull request closed without merging**